### PR TITLE
Set up VFS integration before parsing options

### LINKED ISSUE #9155: owncloud --version reports `Using virtual files plugin: off`
### Expected behaviour
`owncloud --version` should report `Using virtual files plugin: suffix`

### Actual behaviour
`owncloud --version` reports `Using virtual files plugin: off`

### Steps to reproduce
1. `owncloud --version`

### Screenshot
Screenshot from 2021-10-15 15-43-13

### Server configuration
Web server: https://owncloud.tuwien.ac.at

ownCloud version: https://owncloud.tuwien.ac.at/status.php

| name | value |
|  -- | -- |
|installed | true|
|maintenance | false|
|needsDbUpgrade | false|
|version | "10.7.0.4"|
|versionstring | "10.7.0"|
|edition | "Enterprise"|<
|productname | "ownCloud TU Wien"|

### Client configuration
Client version: `owncloud --version`
```
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
ownCloud ownCloud 2.9.1 (build 5500) Oct 13 2021 08:34:12
https://github.com/owncloud/client/commit/42fb299cadd56dc5b7f99bb6946f5bed8a53a2eb
Libraries Qt 5.12.10, OpenSSL 1.1.1l  FIPS 24 Aug 2021
Using virtual files plugin: off
fedora-5.14.11-200.fc34.x86_64
```

Operating system: `$ hostnamectl`
```
 Static hostname: lws84.imws.tuwien.ac.at
       Icon name: computer-laptop
         Chassis: laptop
      Machine ID: 96434f58f7644401a3f9d69da903899d
         Boot ID: ce0b799699ae4148ae05550cf1937bb7
Operating System: Fedora 34 (Workstation Edition) 
     CPE OS Name: cpe:/o:fedoraproject:fedora:34
          Kernel: Linux 5.13.15-200.fc34.x86_64
    Architecture: x86-64
 Hardware Vendor: Lenovo
```

OS language:  `$ locale`
```
LANG=en_GB.UTF-8
LC_CTYPE="en_GB.UTF-8"
LC_NUMERIC=de_AT.UTF-8
LC_TIME=de_AT.UTF-8
LC_COLLATE="en_GB.UTF-8"
LC_MONETARY=de_AT.UTF-8
LC_MESSAGES="en_GB.UTF-8"
LC_PAPER=de_AT.UTF-8
LC_NAME="en_GB.UTF-8"
LC_ADDRESS="en_GB.UTF-8"
LC_TELEPHONE="en_GB.UTF-8"
LC_MEASUREMENT=de_AT.UTF-8
LC_IDENTIFICATION="en_GB.UTF-8"
LC_ALL=
```

Qt version used by client package (Linux only, see also Settings dialog):
```
ownCloud 2.9.1 (build 5500) 42fb29 Oct 13 2021 08:34:12
Libraries Qt 5.12.10, OpenSSL 1.1.1l FIPS 24 Aug 2021
Using virtual files plugin: suffix
fedora-5.14.11-200.fc34.x86_64
```

Client package (From ownCloud or distro) (Linux only): `$ dnf list owncloud*`
```
Installed Packages
owncloud-client.x86_64                   2.9.1-5500   @owncloud
owncloud-client-nautilus.x86_64          2.9.1-5500   @owncloud
owncloud-client-nemo.x86_64              2.9.1-5500   @owncloud
owncloud-client-overlays-icons.x86_64    2.9.1-5500   @owncloud
```

Installation path of client:
`/opt/ownCloud/ownCloud/bin/owncloud`

### Logs
see https://github.com/owncloud/client/issues/9052#issuecomment-922250501

### PR DESCRIPTION
**Pull request by fmoc**
Fixes #9155

### CODE REVIEW

File: src/gui/application.cpp [hunk 1]
```diff
@@ -247,6 +247,26 @@ Application::Application(int &argc, char **argv)
...
+    // Check vfs plugins
+    if (Theme::instance()->showVirtualFilesOption() && bestAvailableVfsMode() == Vfs::Off) {
+        qCWarning(lcApplication) << "Theme wants to show vfs mode, but no vfs plugins are available";
+    }
+    if (isVfsPluginAvailable(Vfs::WindowsCfApi))
+        qCInfo(lcApplication) << "VFS windows plugin is available";
+    if (isVfsPluginAvailable(Vfs::WithSuffix))
+        qCInfo(lcApplication) << "VFS suffix plugin is available";
+
+    if (_quitInstance) {
```
Comment (TheOneRing) [inline src/gui/application.cpp:265]: `_quitInstance` will be set in parseOptions.

With the new parser however I'm pretty sure we can get rid of that bool and directly trigger ::quit from the parser?

**Review by TheOneRing — CHANGES_REQUESTED** [review]
⬆️

File: src/gui/application.cpp (hunk 1 above)
Comment (fmoc) [reply src/gui/application.cpp:265]: I should pay attention when moving code. This block is completely unrelated...

File: changelog/unreleased/9155 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bugfix: --version showed incorrect information about VFS support
+
+--version used to always show "Off", even when a VFS plugin was available.
+This has been fixed now.
+
+https://github.com/owncloud/client/pull/9155
```
Comment (TheOneRing) [inline changelog/unreleased/9155:6]: Isn't one supposed to be an issue?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: changelog/unreleased/9155 (hunk 2 above)
Comment (fmoc) [reply changelog/unreleased/9155:6]: It is an issue, but I typed the URL by hand... I'll fix it.

**Comment by fmoc** [comment]
I don't see how the change could make the tests fail, which do not use any of the commandline parsing code. Any ideas?

**Comment by TheOneRing** [comment]
If an account is configured to use vfs and and we fail to load that plugin we crash.
I assume the change influences the plugin loading in the unit test.

**Comment by TheOneRing** [comment]
Looks like it was a hickup

**Pull request merged**